src/azure-cli-core/tests/test_api_profiles.py

```diff
@@ -0,0 +1,126 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import unittest
+import mock
+
+from azure.cli.core.profiles import (get_api_version,
+                                     supported_api_version,
+                                     ResourceType)
+from azure.cli.core.profiles._shared import APIVersionException
+from azure.cli.core.cloud import Cloud
+
+
+class TestAPIProfiles(unittest.TestCase):
+
+    @mock.patch('azure.cli.core._profile.CLOUD', Cloud('TestCloud', profile='myprofile'))
+    def test_get_api_version(self):
+        ''' Can get correct resource type API version '''
+        test_profile = {'myprofile': {ResourceType.MGMT_STORAGE: '2020-10-10'}}
```

Comment:
Let's pick a year beyond `2020` because it is actually very easy to get to that date. Remember in 1970 they thought it is not a big deal for us to get to 2000?